src/components/sidebar/Sidebar.tsx

```diff
@@ -0,0 +1,91 @@
+import { useState } from "react";
+import { ChatRoomPreview } from "../chatRoomPreview/ChatRoomPreview";
+import "./Sidebar.css";
+import showicon from "../../assets/images/right-arrow.png";
+import hideicon from "../../assets/images/arrow-down-sign-to-navigate.png";
+
+interface SidebarProps {
+  groupChatIds: string[];
+  personalChatIds: string[];
+  onChatRoomPreviewClick: (chatRoomId: string) => void;
+}
+
+export const Sidebar = ({
+  groupChatIds,
+  personalChatIds,
+  onChatRoomPreviewClick,
+}: SidebarProps) => {
+  const [showGroupChats, setShowGroupChats] = useState(true);
+  const [showPersonalChats, setShowPersonalChats] = useState(true);
+
+  const renderGroupChatPreviews = () => (
+    <ul>
+      {groupChatIds.map((id) => (
+        <ChatRoomPreview
+          chatRoomId={id}
+          key={id}
+          onChatRoomPreviewClick={() => onChatRoomPreviewClick(id)}
+        />
+      ))}
+    </ul>
+  );
+
+  const renderPersonalChatPreviews = () => (
+    <ul>
+      {personalChatIds.map((id) => (
+        <ChatRoomPreview
+          key={id}
+          chatRoomId={id}
+          onChatRoomPreviewClick={() => onChatRoomPreviewClick(id)}
```

Comment:
use data attributes or extract out in a component and use callback instead of creating a new handler for each on each render